cmake/FindCLN.cmake

```diff
@@ -47,40 +47,87 @@ if(NOT CLN_FOUND_SYSTEM)
+    message(FATAL_ERROR
+      "Compilation of CLN in the MSYS2 environment is not supported."
+    )
+  endif()
 
   set(CLN_VERSION "1.3.7")
+  set(CLN_SO_MAJOR_VER "6")
+  set(CLN_SO_MINOR_VER "0")
+  set(CLN_SO_PATCH_VER "7")
+  set(CLN_SO_VERSION
+    "${CLN_SO_MAJOR_VER}.${CLN_SO_MINOR_VER}.${CLN_SO_PATCH_VER}"
+  )
   string(REPLACE "." "-" CLN_TAG ${CLN_VERSION})
 
   find_program(AUTORECONF autoreconf)
   if(NOT AUTORECONF)
-    message(SEND_ERROR "Can not build CLN, missing binary for autoreconf")
+    message(FATAL_ERROR "Can not build CLN, missing binary for autoreconf")
+  endif()
+
+  set(CLN_INCLUDE_DIR "${DEPS_BASE}/include/")
+  if(BUILD_SHARED_LIBS)
+    set(LINK_OPTS --enable-shared --disable-static)
+    set(CLN_LIBRARIES "${DEPS_BASE}/${CMAKE_INSTALL_LIBDIR}/libcln${CMAKE_SHARED_LIBRARY_SUFFIX}")
+    if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+      set(CLN_BYPRODUCTS
+        <INSTALL_DIR>/${CMAKE_INSTALL_LIBDIR}/libcln${CMAKE_SHARED_LIBRARY_SUFFIX}
+        <INSTALL_DIR>/${CMAKE_INSTALL_LIBDIR}/libcln.${CLN_SO_MAJOR_VER}${CMAKE_SHARED_LIBRARY_SUFFIX}
+      )
+    else()
+      set(CLN_BYPRODUCTS
+        <INSTALL_DIR>/${CMAKE_INSTALL_LIBDIR}/libcln${CMAKE_SHARED_LIBRARY_SUFFIX}
+        <INSTALL_DIR>/${CMAKE_INSTALL_LIBDIR}/libcln${CMAKE_SHARED_LIBRARY_SUFFIX}.${CLN_SO_MAJOR_VER}
+        <INSTALL_DIR>/${CMAKE_INSTALL_LIBDIR}/libcln${CMAKE_SHARED_LIBRARY_SUFFIX}.${CLN_SO_VERSION}
+      )
+    endif()
+  else()
+    set(LINK_OPTS --enable-static --disable-shared)
+    set(CLN_LIBRARIES "${DEPS_BASE}/${CMAKE_INSTALL_LIBDIR}/libcln${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    set(CLN_BYPRODUCTS <INSTALL_DIR>/${CMAKE_INSTALL_LIBDIR}/libcln${CMAKE_STATIC_LIBRARY_SUFFIX})
+  endif()
+
+  set(CONFIGURE_OPTS "")
+  # CLN yields the following message at the end of the build process.
+  #     WARNING: `makeinfo' is missing on your system.
+  # This is a specific issue to Github CI on linux environments.
+  # Since makeinfo just builds the documentation for CLN,
+  # it is possible to get around this issue by just disabling it:
+  set(CONFIGURE_ENV env "MAKEINFO=true")
+
+  if(CMAKE_CROSSCOMPILING OR CMAKE_CROSSCOMPILING_MACOS)
+    set(CONFIGURE_OPTS
+      --host=${TOOLCHAIN_PREFIX}
+      --build=${CMAKE_HOST_SYSTEM_PROCESSOR})
+
+    set(CONFIGURE_ENV ${CONFIGURE_ENV} ${CMAKE_COMMAND} -E
+      env "CC_FOR_BUILD=cc")
+    if (CMAKE_CROSSCOMPILING_MACOS)
+      set(CONFIGURE_ENV
+        ${CONFIGURE_ENV}
+        env "CFLAGS=--target=${TOOLCHAIN_PREFIX}"
+        env "LDFLAGS=-arch ${CMAKE_OSX_ARCHITECTURES}")
+    endif()
   endif()
 
   ExternalProject_Add(
     CLN-EP
     ${COMMON_EP_CONFIG}
-    URL "https://www.ginac.de/CLN/cln.git/?p=cln.git\\\;a=snapshot\\\;h=cln_${CLN_TAG}\\\;sf=tgz"
-    URL_HASH SHA1=bd6dec17cf1088bdd592794d9239d47c752cf3da
+    URL "https://www.ginac.de/CLN/cln-1.3.7.tar.bz2"
+    URL_HASH SHA1=17cf2c60e262e30f57caae39692fce7917e11d95
     DOWNLOAD_NAME cln.tgz
```

Comment:
This suffix doesn't match the downloaded file (bz2 vs. tgz).

Reply:
Good catch! I didn't notice since CMake doesn't rely on the filename extension.